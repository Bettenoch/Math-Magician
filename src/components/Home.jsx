import React from 'react';
import '../styles/Home.scss';

const Home = () => (
  <div className="home-container">
    <article>
      <h2>Welcome to our page!</h2>
    </article>
    <article className="home-intro">
      <p>
        Math Magician&apos;s sole focus is to provide fast, comprehensive, convenient,
        free online calculators in a plethora of areas.
        Currently, we have around 200 calculators to help you &quot; do the math&quot;
        quickly in areas
        such as finance,
        fitness, health, math, and others,
        and we are still developing more. Our goal is to become the one-stop,
        go-to site for people who need to make quick calculations.
      </p>
      <p>
        Additionally, we believe the internet should be a source of free information.
        Therefore, all of our tools and services are completely free, with no registration required.
        We coded and developed each calculator individually and
        put each one through strict, comprehensive testing.
        However, please inform us if you notice even the slightest error â€“
        your input is extremely valuable to us. While most calculators on
        Calculator.net are designed to be universally applicable for worldwide usage,
        some are for specific countries only.
        For example, the Income Tax Calculator is for United States residents only.
      </p>
    </article>
  </div>
);

export default Home;
