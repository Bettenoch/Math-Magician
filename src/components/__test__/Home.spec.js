import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../Home';

describe('Home snapshots', () => {
  it('should match the Home DOM snapshots', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test Home content', () => {
  it('should match the text content', () => {
    render(<Home />);
    const welcome = screen.getByTestId('welcome-page');
    expect(welcome).toHaveTextContent('Welcome to our page!');
  });
});
