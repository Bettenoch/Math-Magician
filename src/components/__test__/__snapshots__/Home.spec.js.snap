// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshots should match the Home DOM snapshots 1`] = `
<div
  className="home-container"
>
  <article>
    <h2
      data-testid="welcome-page"
    >
      Welcome to our page!
    </h2>
  </article>
  <article
    className="home-intro"
  >
    <p>
      Math Magician's sole focus is to provide fast, comprehensive, convenient, free online calculators in a plethora of areas. Currently, we have around 200 calculators to help you " do the math" quickly in areas such as finance, fitness, health, math, and others, and we are still developing more. Our goal is to become the one-stop, go-to site for people who need to make quick calculations.
    </p>
    <p>
      Additionally, we believe the internet should be a source of free information. Therefore, all of our tools and services are completely free, with no registration required. We coded and developed each calculator individually and put each one through strict, comprehensive testing. However, please inform us if you notice even the slightest error â€“ your input is extremely valuable to us. While most calculators on Calculator.net are designed to be universally applicable for worldwide usage, some are for specific countries only. For example, the Income Tax Calculator is for United States residents only.
    </p>
  </article>
</div>
`;
